pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -71,12 +71,18 @@ func (s *CloudWatchService) Init() error {
 }
 
 func (s *CloudWatchService) NewExecutor(*models.DataSource) (plugins.DataPlugin, error) {
-	return newExecutor(s.LogsService), nil
+	config := setting.GetCfg()
+	return newExecutor(s.LogsService, newAWSSettings(config.AWSAllowedAuthProviders, config.AWSAssumeRoleEnabled)), nil
 }
 
-func newExecutor(logsService *LogsService) *cloudWatchExecutor {
+func newAWSSettings(allowedAuthProviders []string, assumeRoleEnabled bool) *awsSettings {
```

Comment:
Why not just pass `*setting.Cfg` to the executor? It makes for easier and more change friendly code.

Reply:
Yeah I get that, but it does make for more change resistant code, i.e. there will be more work for maintainers when more (or less) of the settings have to be used. It also increases the potential for bugs, since you have to replicate fields across types.

I'm also thinking of the potential future case of supporting updating of configuration parameters at runtime. I don't know if that will ever be a requirement, but it won't work if parameters are copied around instead of the original config.

Reply:
> I'm also thinking of the potential future case of supporting updating of configuration parameters at runtime. I don't know if that will ever be a requirement, but it won't work if parameters are copied around instead of the original config.

Good point. I'll change